src/main.cpp

```diff
@@ -2784,6 +2786,42 @@ void static UpdateTip(CBlockIndex *pindexNew) {
       Checkpoints::GuessVerificationProgress(chainParams.Checkpoints(), chainActive.Tip()), pcoinsTip->DynamicMemoryUsage() * (1.0 / (1<<20)), pcoinsTip->GetCacheSize());
 
     cvBlockChange.notify_all();
+    
+    
+
+    // MVHF-BU
+    const int MVHF_FORK_BLOCK = 50000;
+
+    //// 	Auto Backup At Block 		////
+    // Test autobackupwalletpath argument
+    if (!fAutoBackupDone)
+    {
+    	std::string strWalletBackupFile = GetArg("-autobackupwalletpath", "");
+
+		int BackupBlock = GetArg("-autobackupblock", MVHF_FORK_BLOCK - 1);
+
+		//LogPrintf("DEBUG: autobackupwalletpath=%s\n",strWalletBackupFile);
+		//LogPrintf("DEBUG: autobackupblock=%d\n",BackupBlock);
+
+		if(strWalletBackupFile != "") {
+
+			if (GetBoolArg("-disablewallet", false)) {
+				LogPrintf("-disablewallet and -autobackupwalletpath conflict so automatic backup disabled.");
+				fAutoBackupDone = true;
+			}
+			else
+				// Auto Backup defined so check block height
+				if (chainActive.Height() >= BackupBlock )
+				{
+					if (GetMainSignals().BackupWalletAuto(strWalletBackupFile, BackupBlock))
+						fAutoBackupDone = true;
+					else
+						throw std::runtime_error("CWallet::BackupWalletAuto() : Auto wallet backup failed!");
```

Comment:
Is the runtime error caught handled anywhere?
Grepping through the code I see a number of shutdown-related functions.
Perhaps it would be better to Flush() the wallet and tell the main threads to shutdown and exit gracefully?
I'm going to merge the PR as is, we can have a look at the details of this later on.
Especially the test framework not being able to catch the expected shutdown condition (if implemented using a runtime error) is unfortunate. Perhaps we can fix this later.